zipline/pipeline/graph.py

```diff
@@ -130,8 +130,38 @@ def initial_refcounts(self):
         refcounts = self.out_degree()
         for t in self.outputs.values():
             refcounts[t] += 1
+
+        for t in initial_workspace:
+            self.decref_dependencies(t, refcounts)
+
         return refcounts
 
+    def decref_dependencies(self, term, refcounts):
```

Comment:
shouldn't decref release the data? It seems like this makes it easier to get the refcounts and workspace out of sync.